mongo/job_store.go

```diff
@@ -173,3 +175,29 @@ func (m *JobStore) GetJob(ctx context.Context, id string) (models.Job, error) {
 func (m *JobStore) Checker(ctx context.Context, state *healthcheck.CheckState) error {
 	return m.healthClient.Checker(ctx, state)
 }
+
+// PutNumberOfTasks updates the number_of_tasks in a particular job, from the collection, specified by its id
+func (m *JobStore) PutNumberOfTasks(ctx context.Context, id string, numTasks int) (err error) {
+	s := m.Session.Copy()
+	defer s.Close()
+	log.Event(ctx, "putting number of tasks", log.Data{"id": id, "numTasks": numTasks})
+
+	updates := make(bson.M)
+	updates["number_of_tasks"] = numTasks
+	updates["last_updated"] = time.Now()
+	err = m.UpdateJob(updates, s, id)
+
+	return err
+}
+
+// UpdateJob updates a particular job with the values passed in through the 'updates' input parameter
+func (m *JobStore) UpdateJob(updates bson.M, s *mgo.Session, id string) error {
+	update := bson.M{"$set": updates}
+	if err := s.DB(m.Database).C(m.Collection).UpdateId(id, update); err != nil {
+		if err == mgo.ErrNotFound {
+			return apierrors.ErrJobNotFound
```

Comment:
This error should belong to this package - the store package shouldn't depend on the API as this will create a cyclic dependency.

Reply:
ok - changed now, thanks